### PR TITLE
Fix Tril Triu SymInt

### LINKED ISSUE #165613: Tril Does not Support SymInt
### 🐛 Describe the bug

If torch.export a module containing tril with dynamic shape, there will be tril related symbol failure

PS: Triu most likely has the same issue as well

### Versions

2.9, 2.8, 2.7.1

cc @chauhang @penguinwu @ezyang @bobrenjc93 @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

### PR DESCRIPTION
**Pull request by parsshar-RH**
Fixes #165613 

### Summary:

- This MR fixes an issue where `torch.tril `and `torch.triu` with dynamic diagonal values cause torch.export to incorrectly infer unnecessary constraints between dynamic dimensions.
-  Ensured proper SymInt type annotations for diagonal parameter
-  Updated C++ implementation to correctly handle SymInt diagonal values.

### Impacts:
module: dynamic shapes

cc @ezyang @penguinwu @bobrenjc93

### CODE REVIEW

File: aten/src/ATen/functorch/BatchRulesViews.cpp [hunk 1]
```diff
@@ -534,20 +534,20 @@ Tensor trace_decomp(const Tensor& tensor) {
...
 
 std::tuple<Tensor, std::optional<int64_t>> triu_batch_rule(
     const Tensor& self,
     std::optional<int64_t> self_bdim,
-    int64_t diagonal = 0) {
+    c10::SymInt diagonal = 0) {
   TORCH_CHECK(self.dim() >= 2, "triu: The input tensor must have at least 2 dimensions.");
   auto self_ = moveBatchDimToFront(self, self_bdim);
-  auto result = at::triu(self_, diagonal);
+  auto result = at::triu_symint(self_, diagonal);
```
Comment (Skylion007) [inline aten/src/ATen/functorch/BatchRulesViews.cpp]:
```suggestion
  auto result = at::triu_symint(self_, std::move(diagonal));

File: aten/src/ATen/functorch/BatchRulesViews.cpp [hunk 2]
```diff
@@ -534,20 +534,20 @@ Tensor trace_decomp(const Tensor& tensor) {
 std::tuple<Tensor, std::optional<int64_t>> tril_batch_rule(
     const Tensor& self,
     std::optional<int64_t> self_bdim,
-    int64_t diagonal = 0) {
+    c10::SymInt diagonal = 0) {
   TORCH_CHECK(self.dim() >= 2, "tril: The input tensor must have at least 2 dimensions.");
   auto self_ = moveBatchDimToFront(self, self_bdim);
-  auto result = at::tril(self_, diagonal);
+  auto result = at::tril_symint(self_, diagonal);
```
Comment (Skylion007) [inline aten/src/ATen/functorch/BatchRulesViews.cpp]:
```suggestion
  auto result = at::tril_symint(self_, std::move(diagonal));
```

**Review by Skylion007 — COMMENTED** [review]
Super minor nits. Need to double check if tri[u]_symint takes SymInt by value or not

File: aten/src/ATen/functorch/BatchRulesViews.cpp (hunk 2 above)
Comment (parsshar-RH) [reply aten/src/ATen/functorch/BatchRulesViews.cpp]: @Skylion007 

Confirmed that tril_symint takes SymInt by value

Thanks for the review

File: aten/src/ATen/functorch/BatchRulesViews.cpp (hunk 1 above)
Comment (parsshar-RH) [reply aten/src/ATen/functorch/BatchRulesViews.cpp]: @Skylion007

Confirmed that triu_symint takes SymInt  by value

Thanks for the review

**Review by ezyang — APPROVED** [review]

**Review by Skylion007 — APPROVED** [review]
